internal/controller/pkg/revision/establisher.go

```diff
@@ -79,7 +79,7 @@ func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, p
 
 		// Make a copy of the desired object to be populated with existing
 		// object, if it exists.
-		current := res.DeepCopyObject()
+		current := d
 		err := e.client.Get(ctx, types.NamespacedName{Name: d.GetName(), Namespace: d.GetNamespace()}, current)
```

Comment:
`e.client.Get` has been refactored and now expects a `client.Object` rather than `runtime.Object` as last argument, as such using `res.DeepCopyObject()` (which returns a `runtime.Object`) is not possible. I've kept `current` around because it's used in line `103` too, but in reality both of them can be changed to use `d` and `current` can be removed altogether.

Reply:
@khos2ow unfortunately, I don't think this can be simplified in this manner. `res` is an interface here, which may have an underlying pointer type. In that case, instantiating `current` with `d` and then "getting into" `current` will actually change the underlying data of `d` as well. It looks like we will have to attempt to assert `current` to `client.Object` after the deep copy.

Here is a small toy example to illustrate the situation: https://play.golang.org/p/VeK7BHiKsad